index.bs

```diff
@@ -8923,6 +8930,79 @@ The [=remote end steps=] given |session|, and |command parameters| are:
 
 </div>
 
+#### The input.setFiles Command ####  {#command-input-setFiles}
+
+The <dfn export for=commands>input.setFiles</dfn> command sets a given file
+input with a given set of file paths.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      input.SetFiles = (
+        method: "input.setFiles",
+        params: input.SetFilesParameters
+      )
+
+      input.SetFilesParameters = {
+        context: browsingContext.BrowsingContext,
+        element: script.SharedReference,
+        files: [*text]
+      }
+    </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl">
+     EmptyResult
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for input.setFiles">
+
+The [=remote end steps=] given |session|, and |command parameters| are:
+
+1. Let |context id| be the value of the <code>context</code> field of
+   |command parameters|.
+
+1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+   with |context id|.
+
+1. Let |top-level context| be |context|'s [=top-level browsing context=].
+
+<!-- converting to realm here is kind of silly since we immediately convert back
+      in [=deserialize remote reference=] -->
+1. Let |document| be |top-level context|'s [=active document=].
```

Comment:
why do we need to access the document of the top-level browsing context?

Reply:
it sounds like environment settings can be retried from the `|context|'s [=active document=].`?